### PR TITLE
Execute SQL Operation now sends notification

### LINKED ISSUE #339: ExecuteSQL Operation does not send notifications
ExecSQL Operation should send notifications about changes if `RawQuery` contains `affectedTables`.

### PR DESCRIPTION
**Pull request by artem-zinnatullin**
Closes #339.

@nikitin-da PTAL

### CODE REVIEW

**Comment by nikitin-da** [comment]
retest this please

File: storio-sqlite/src/test/java/com/pushtorefresh/storio/sqlite/operation/execute/PreparedExecuteSQLTest.java [hunk 1]
```diff
@@ -27,16 +50,67 @@ public void blocking() {
...
+            this.queryWithNotification = queryWithNotification;
 
-        Stub() {
             storIOSQLite = mock(StorIOSQLite.class);
             internal = mock(StorIOSQLite.Internal.class);
-            rawQuery = new RawQuery.Builder().query("DROP TABLE users").build();
+
+            if (queryWithNotification) {
+                rawQuery = new RawQuery.Builder()
+                        .query("DROP TABLE users!")
```
Comment (nikitin-da) [inline storio-sqlite/src/test/java/com/pushtorefresh/storio/sqlite/operation/execute/PreparedExecuteSQLTest.java]: With expression!

**Comment by nikitin-da** [comment]
My stable test =(

**Comment by artem-zinnatullin** [comment]
retest this please

**Comment by artem-zinnatullin** [comment]
Probably, it's my fault, but locally tests pass normally. Uh, flaky tests…

**Comment by artem-zinnatullin** [comment]
Damn, I figured out the problem of this test (probably, other tests can be broken too):

1. Fill db with some data
2. Perform Get Operation and observe results
3. Delete some data (here is the problem)
4. Check that first result of Get Operation was ["full dataset"], then ["dataset - deletion1"], ["dataset - deletion1 - deletion2", …]

Problem Get Operation is asynchronous and it can be executed AFTER deletion, so you won't see ["full dataset"]!

Also possible situation:

1. Fill db with some data
2. Perform Get Operation and observe results
3. Delete some data (okay for first emission, but Get Operation is async and it can skip some states between deletions)

Solution: wait for first emission before deletions, wait for emission after each deletion.

**Comment by artem-zinnatullin** [comment]
I filed #355 for that, @nikitin-da can you PTAL and merge this PR if everything is okay except flaky tests?

**Comment by nikitin-da** [comment]
Yes, of cause

**Comment by artem-zinnatullin** [comment]
Typo in "of cause" :)

**Comment by nikitin-da** [comment]
ヽ(ﾟ〇ﾟ)ﾉ

**Comment by nikitin-da** [comment]
Of course I meant "of course"

**Comment by artem-zinnatullin** [comment]
Of course :)
2 июня 2015 г. 12:11 AM пользователь "Dmitrii Nikitin" <
notifications@github.com> написал:

**Pull request merged**